homeassistant/components/growatt/sensor.py

```diff
@@ -0,0 +1,178 @@
+"""
+Support for Growatt Plant energy production sensors.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.growatt/
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ["growatt==0.0.2"]
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }
+)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Growatt Plant sensor."""
+    import growatt
+
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    growatt_client = growatt.GrowattApi()
+
+    is_login_success = login(growatt_client, username, password)
+    if is_login_success:
+        sensor_today = GrowattPlantToday(
+            hass, growatt_client, username, password
+        )
+        sensor_total = GrowattPlantTotal(
+            hass, growatt_client, username, password
+        )
+        sensor_current = GrowattPlantCurrent(
+            hass, growatt_client, username, password
+        )
+        add_entities([sensor_today, sensor_total, sensor_current])
+        return True
+
+    return False
+
+
+def login(client, username, password):
+    """Login to the growatt server."""
+    import growatt
+
+    try:
+        client.login(username, password)
+        return True
+    except growatt.LoginError as error:
+        logging.error(error)
+        return False
+
+
+class GrowattPlant(Entity):
+    """Base Growatt sensor class."""
+
+    def __init__(self, hass, client, username, password):
+        """Initialize the sensor."""
+        self._hass = hass
+        self._client = client
+        self._username = username
+        self._password = password
+        self._state = None
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._state
+
+    @staticmethod
+    def convert_multiplier(metric_name, multiplier_lookup):
+        """Convert a value to a given multiplier."""
+        if metric_name not in multiplier_lookup:
+            message = (
+                "Found an unsupported metric name {}"
+                "cannot convert safely to kWh."
+            ).format(metric_name)
+            logging.error(message)
+            raise ValueError(message)
+        return multiplier_lookup[metric_name]
+
+
+class GrowattPlantTotals(GrowattPlant):
+    """Representation of a Growatt plant sensor."""
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit this state is expressed in."""
+        return "kWh"
+
+    def _convert_to_kwh(self, value: str, metric_name: str):
```

Comment:
Do we have util method to do that?

Reply:
I did not find one. I seems worth to have it. I'll implement as part of a new PR.